src/sentry/scim/endpoints/utils.py

```diff
@@ -0,0 +1,138 @@
+import sentry_sdk
+from rest_framework.exceptions import ParseError
+from rest_framework.negotiation import BaseContentNegotiation
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.api.paginator import BadPaginationError, GenericOffsetPaginator
+from sentry.models import AuthProvider
+from sentry.utils.cursors import Cursor
+
+from .constants import SCIM_API_LIST
+
+
+class IgnoreClientContentNegotiation(BaseContentNegotiation):
+    # TODO: validate scim content type
+    def select_parser(self, request, parsers):
+        """
+        Select the first parser in the `.parser_classes` list.
+        """
+        return parsers[0]
+
+    def select_renderer(self, request, renderers, format_suffix):
+        """
+        Select the first renderer in the `.renderer_classes` list.
+        """
+        return (renderers[0], renderers[0].media_type)
+
+
+class OrganizationSCIMPermission(OrganizationPermission):
+    scope_map = {
+        "GET": ["member:read", "member:write", "member:admin"],
+        "POST": ["member:write", "member:admin"],
+        "PATCH": ["member:write", "member:admin"],
+        "PUT": ["member:write", "member:admin"],
+        "DELETE": ["member:admin"],
+    }
+
+    def has_object_permission(self, request, view, organization):
+        result = super().has_object_permission(request, view, organization)
+        # The scim endpoints should only be used in conjunction with a SAML2 integration
+        if not result:
+            return result
+        try:
+            auth_provider = AuthProvider.objects.get(organization=organization)
+        except AuthProvider.DoesNotExist:
+            return False
+        if not auth_provider.flags.scim_enabled:
+            return False
+
+        return True
+
+
+class SCIMEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationSCIMPermission,)
+    content_negotiation_class = IgnoreClientContentNegotiation
+
+    def paginate(
```

Comment:
had to define a custom pagination method for SCIM as it wants its serialized data wrapped in a bit of JSON, and to account for SCIM cursors being 1 indexed